### PR TITLE
:recycle: Implemented Dependency Injection

### PR DESCRIPTION
**Pull request by VujicGit**

### CODE REVIEW

File: Backend/Users/Service/FeedbackService.cs [hunk 1]
```diff
@@ -1,26 +1,76 @@
...
+        public IEnumerable<Feedback> GetAllApprovedFeedback()
+        {
+            return feedbackRepository.GetAllApprovedFeedback();
+        }
+
+        public bool UpdateStatus(int feedbackId, bool status)
+        {
+            return feedbackRepository.UpdateStatus(feedbackId, status);
+        }
+        public Feedback CreateFeedback(string userId, string additionalNotes, Boolean anonymous, Boolean allowed)
```
Comment (jelena-vlajkov) [inline Backend/Users/Service/FeedbackService.cs:35]: Change methods that receive more than 3 parameters from controllers to DTOs.

File: Backend/Users/TableBuilder/Interfaces/IPatientTableBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        Address SaveAddress(string street, int number, int floor, int apartment, int cityId);
+        State SaveState(string name);
+        State BuildSate(string name);
+        City SaveCity(int id, string name, long stateId);
+        City BuildCity(int id, string name, long stateId);
+        InsurancePolicy BuildInsurancePolicy(string id, string company, DateTime startPeriod, DateTime endPeriod);
+        InsurancePolicy SavePolicy(string id, string Company, DateTime startTime, DateTime endTime);
+        MedicalRecord BuildMedicalRecord(string patientId, string condition, string bloodType);
+        MedicalRecord SaveMedicalRecord(string patientId, string condition, string bloodType);
+    }
```
Comment (jelena-vlajkov) [inline Backend/Users/TableBuilder/Interfaces/IPatientTableBuilder.cs:21]: Change methods that receive more than 3 parameters from controllers to DTOs.

File: WebApplication/Controller/PrescriptionController.cs [hunk 3]
```diff
@@ -35,7 +35,7 @@ public IActionResult GetSearchedPrescription(PrescriptionSearchDTO dto)
                 return BadRequest(e.Message);
             }
 
-            List<Prescription> prescriptions = controller.GetSearchedReports(dto.Medicine, dto.HourlyIntake, dto.StartDate, dto.EndDate);
+            List<Prescription> prescriptions = _prescriptionSearchService.GetSearchedPrescription(dto.Medicine, dto.HourlyIntake, dto.StartDate, dto.EndDate);
```
Comment (jelena-vlajkov) [inline WebApplication/Controller/PrescriptionController.cs:38]: Change methods that receive more than 3 parameters from controllers to DTOs.

File: WebApplication/Controller/SurveyController.cs [hunk 4]
```diff
@@ -17,26 +18,25 @@ namespace WebApplication.Controller
...
         }
 
         [HttpPost("createSurvey")]
         public IActionResult Post(PostSurveyDTO postSurveyDTO)
         {
-
-            SurveyService surveyService = new SurveyService();           
-            Survey surveySuccessfullyCreated = surveyContoller.CreateSurvey(postSurveyDTO.surveyQuestions, postSurveyDTO.surveyAnswers, postSurveyDTO.appointmentId);
+         
+            Survey surveySuccessfullyCreated = _surveyService.CreateSurvey(postSurveyDTO.surveyQuestions, postSurveyDTO.surveyAnswers, postSurveyDTO.appointmentId);
```
Comment (jelena-vlajkov) [inline WebApplication/Controller/SurveyController.cs:39]: Change methods that receive more than 3 parameters from controllers to DTOs.

**Review by jelena-vlajkov — APPROVED** [review]
Please change the methods that receive more than 2 parameters from controllers to DTOs.

**Review by oakenknight — APPROVED** [review]

**Pull request merged**